zigpy/endpoint.py

```diff
@@ -80,6 +80,26 @@ async def initialize(self):
 
         self.status = Status.ZDO_INIT
 
+    def initialize_from_quirk(self, endpoint):
+        if self.status == Status.ZDO_INIT:
+            return
+        self.profile_id = endpoint["profile_id"]
+        self.device_type = endpoint["device_type"]
+        if self.profile_id == 260:
+            self.device_type = zigpy.profiles.zha.DeviceType(self.device_type)
+        elif self.profile_id == 49246:
+            self.device_type = zigpy.profiles.zll.DeviceType(self.device_type)
+
+        for cluster in endpoint["input_clusters"]:
+            clus = self.add_input_cluster(cluster)
+            if cluster == 0:
+                clus._attr_cache[4] = self.manufacturer
+                clus._attr_cache[5] = self.model
+        for cluster in endpoint["output_clusters"]:
+            self.add_output_cluster(cluster)
+
+        self.status = Status.ZDO_INIT
+
```

Comment:
Nope. Make a `module` method or `CustomDevice` class method to create a device instance from a quirk.
You give it a zigpy device instance and a model and it returns a  Quirk instance. Similar to https://github.com/zigpy/zha-device-handlers/blob/347b5084bf1cbc767e71f916087b6437b3b2d1c8/tests/conftest.py#L77

make this in a second PR